imports/plugins/core/dashboard/client/components/toolbar.js

```diff
@@ -113,7 +122,7 @@ class PublishControls extends Component {
   }
 
   renderCustomControls() {
-    if (this.props.dashboardHeaderTemplate) {
+    if (this.props.dashboardHeaderTemplate && this.props.hasCreateProductAccess) {
```

Comment:
Not specific to this PR, but has the `createProduct` permission become a bit of a universal check for more than just creating products?